py/torch_tensorrt/dynamo/lowering/_decompositions.py

```diff
@@ -174,6 +174,19 @@ def empty_permuted_decomposition(*args, **kwargs) -> torch.Tensor:
     return torch.empty([empty_size[l] for l in empty_permute], **kwargs).permute(perm)
 
 
+@register_torch_trt_decomposition(
+    torch.ops.aten.scatter_add.default, registry=TORCH_TRT_DECOMPOSITIONS
+)
+def scatter_add_decomposition(
+    input_tensor: torch.Tensor,
+    src_tensor: torch.Tensor,
+    dim: int,
+    index: torch.Tensor,
+) -> torch.Tensor:
+    input_tensor_to_add = torch.scatter(torch.empty_like(input_tensor), dim, index, src_tensor)
+    scatter_add_tensor = torch.add(input_tensor, input_tensor_to_add.cuda())
```

Comment:
Without `.cuda()` does the decomposition fail?

Reply:
Yes in this case it complains that the two tensors are on different devices.